cunumeric/_ufunc/ufunc.py

```diff
@@ -262,37 +321,132 @@ def __call__(
 
         return out
 
-    def __repr__(self):
-        return f"<ufunc {self._name}>"
-
 
-class binary_ufunc(ufunc):
-    def __init__(self, name, doc, op_code, types, red_code=None):
-        self._name = name
+class multiout_unary_ufunc(ufunc):
+    def __init__(self, name, doc, op_code, types):
+        super().__init__(name, doc, types)
         self._op_code = op_code
-        self._types = types
         self._resolution_cache = {}
-        self._red_code = red_code
-        self.__doc__ = doc
 
-    @property
-    def nin(self):
-        return 2
+    def _resolve_dtype(self, arr, casting, precision_fixed):
+        if arr.dtype.char in self._types:
+            return arr, to_dtypes(self._types[arr.dtype.char])
 
-    @property
-    def nout(self):
-        return 1
+        if arr.dtype in self._resolution_cache:
```

Comment:
I think you should never enter this check if `precision_fixed` is true; instead you should fail immediately if the previous check fails. Otherwise you might pick up an entry on the resolution cache that was previously made for the same dtype, but with `precision_fixed == False`.

Reply:
should be addressed